File: Army/units/berserk.cpp
#include "berserk.h"

using namespace std;

Berserk::Berserk(const string& name, int healthPoint, int damage) : Unit(name, healthPoint, damage)  {
    setAbility(new BerserkAbility(this));
    setCurrentState(new HumanState(this));
    setNextState(new HumanState(this));
    setUnitType(berserk);
}

Berserk::~Berserk() {}

void Berserk::takeMagicDamage(int damage) {}

File: Java/src/com/gymfox/army/units/Berserk.java
package com.gymfox.army.Units;

import com.gymfox.army.Ability.BerserkAbility;

public class Berserk extends Unit {
    public Berserk(String name, int healthPointLimit, int damage) {
        super(name, healthPointLimit, damage);
        this.ability = new BerserkAbility(this);
        this.immunityToMagic = true;
    }
}
